network/src/test/java/info/metadude/android/eventfahrplan/network/temporal/DateParserTest.kt

```diff
@@ -1,24 +1,43 @@
 package info.metadude.android.eventfahrplan.network.temporal
 
 import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatCode
 import org.junit.Test
 
 class DateParserTest {
 
     @Test
     fun `getDateTime date and time`() {
         assertThat(DateParser.getDateTime("2019-01-01T00:00:00Z")).isEqualTo(1546300800000)
+		assertThat(DateParser.getDateTime("1970-01-01T00:00:00Z")).isEqualTo(0)
+		assertThat(DateParser.getDateTime("1970-01-01T00:00:00+00:00")).isEqualTo(0)
+		assertThat(DateParser.getDateTime("1970-01-01T02:00:00+01:00")).isEqualTo(3600000)
+		
+		// test all formats of datetimes ever seen in schedule.xml files
+		assertThat(DateParser.getDateTime("2016-09-14T14:30:00+02:00")).isEqualTo(1473856200000)
+		assertThat(DateParser.getDateTime("2016-09-14T14:30:00+0200")) .isEqualTo(1473856200000)
+		assertThat(DateParser.getDateTime("2016-09-14T12:30:00Z"))     .isEqualTo(1473856200000)
+
+		// testing result of local time parsing for equality is not possible because result depends on execution environment:
+		assertThatCode(() -> DateParser.getDateTime("1970-01-01T03:00:00")).doesNotThrowAnyException()
```

Comment:
This date would be parseable in the following format (mind the trailing `Z`):

``` kotlin
assertThat(DateParser.getDateTime("1970-01-01T03:00:00Z")).isEqualTo(10800000)
```

Would that be a sufficient test case for you?

Reply:
As it resulted in the conversation of #102 this is something that happens in real-life scenarios, so it should work as-is.

Z would be semantically wrong, as such a string is defined to be a "local" datetime (which is of course bad in something like a schedule.xml, which does not provide location information at any other place). Just as the comment states. Therefore the result can not be checked for equality, because e.g. on a german system the result should be 7200000 while on an UK system it is indeed 10800000. I dont know if you have control over the env settings of the test environment.

In my opinion, this format is a better candidate for the question wether it should be supported at all. Not supporting this can be reasoned easily with "times without zone info are bad".

I can also provide feedback in which schedules these appeared.

Reply:
I agree with you that such a time format is a real-life scenario. I however believe that the app should **not** make any **assumptions** about the time zone offset provided in the _schedule.xml_. I have run into errors myself several times because of this. I think we are better off with throwing a descriptive error as early as possible - and in such a case to reach out to the backend people to let them fix it.

This is related to the changes you push in #102.

Reply:
Full ACK that guessing timezone is evil and has lead to many things in the past. It might basically work because attendants are usually at the same physical place and therefore the device probably has the correct interpretation of "local" time, but now with Corona.... and well, it's valid XML / ISO.

Regarding the formats I did the research:

Offsets without colon (e.g. ends with +0100 instead of +01:00) happens only in VERY old schedules, I found it in GPN15 and EP14, both do not have a generator tag or information, so probably those were generated in a custom way.

Unfortunately, local datetimes appear in tons of schedules up until recently, e.g. the DiVoC schedule.
So you were right that the format witihout colons probably does not need any care (though it would be pretty easy).

But local datetimes seem to be a thing. Refusing those would probably lead to many event visitors not being able to import the schedule. Hmpf. I can give you a list of XML URLs or the content privately, I dont want to have them visible in public.

Reply:
Make the time zone and whether or not schedules may contain a local datetime part of the flavor configuration. That way the code doesn't have to make assumptions.

Reply:
Yes, I thought about adding time zone offset information to the flavor configuration, too. That would also allow to adjust the `ConferenceTimeFrame` (used for automatic updates) which does not have these information at the moment. Nevertheless, there is no guarantee that what is configured in the flavor is what we receive from the backend as past experience taught me. At the moment I plead for making time zone offset a requirement to avoid issues or workaround in the app.

Reply:
Is that flavor configuration per conference? Then it should be a viable option.

If you decide to not support local datetimes, there should be a very accurate error message, since based on the presence in the schedules, it will hit the users a lot. This means, the format needs special treatment anyway (e.g. dedicated exception class)

I'm currently researching, which tools generated the recent schedules with local datetimes.

Reply:
> ... since based on the presence in the schedules, it will hit the users a lot

Why do you think there are many schedules without time zone offset information?

> I'm currently researching, which tools generated the recent schedules with local datetimes.

Please let me know. By now, I am aware of Pentabarf, Frab and Pretalx as the backends which I regularly work with.

:speaking_head: At this point it might make sense to take the discussion into a meeting. Let me know if you are interested to talk about the direction I want to take the app development.

Reply:
My assumption is based on ~60 events out of a total 264 events in the VOC tracker have local datetimes in them.
But first research shows that probably most of them do not come from Frab, Pentabarf etc. The only thing that should be changed is that obviously schedules produced by voctosched also have local datetimes, and that seems to account for most of the recent occurences.

The interest in the app development is present, but not sufficient to invest more time into this. My main concern is to "just get the datetime stuff right", because of the problems in the past and because the schedule.xml is something I have to deal with very often.

Reply:
Now you got me even more curios. What is your "very often" involvement in schedule topics? Take it offline at `[AT]tbsprs[COLON]matrix[DOT]org` or via my email address which you find in the commits.

Reply:
I refactored the test cases now - after we talked in person about your ideas on these tests. Please take a look at e69b24c5cc278b32c5d86115e9a968af9d9c1db4 and let me know if these fit for you. I changed the input in `getDayChange returns minutes of a day for first CEST minute of 2020 leap year` - the remaining should still be your values. Please let me know if the test descriptions make sense to you.

Reply:
I commented on the commit, I hope this is okay.

Reply:
Thanks. I changed the commit and replied.

Reply:
Ok, this is in really good shape now IMO!